fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/recordrepair/RecordValidationTest.java

```diff
@@ -345,40 +369,41 @@ void testValidateRecordCorruptVersion(int formatVersion) throws Exception {
             RecordValidator valueValidator = new RecordValueValidator(store);
             // Currently, we don't handle corrupt versions.
-            Exception ex = assertThrows(ExecutionException.class, () -> valueValidator.validateRecordAsync(result.get(0).getPrimaryKey()).get());
-            assertTrue(ex.getCause() instanceof UnknownValidationException);
-            ex = assertThrows(ExecutionException.class, () -> valueValidator.validateRecordAsync(result.get(1).getPrimaryKey()).get());
-            assertTrue(ex.getCause() instanceof UnknownValidationException);
+            result.forEach(rec -> {
+                Exception ex = assertThrows(ExecutionException.class, () -> valueValidator.validateRecordAsync(rec.getPrimaryKey()).get());
+                assertTrue(ex.getCause() instanceof UnknownValidationException);
+            });
 
             commit(context);
         }
     }
 
-    private static void validateRecordVersion(final FDBRecordStore store, final Tuple primaryKey, final @Nonnull String expectedVersionValidationCode) throws InterruptedException, ExecutionException {
-        RecordValidator versionValidator = new RecordVersionValidator(store);
-        final RecordValidationResult versionValidatorResult = versionValidator.validateRecordAsync(primaryKey).get();
-        if (expectedVersionValidationCode.equals(RecordValidationResult.CODE_VALID)) {
-            assertTrue(versionValidatorResult.isValid());
-        } else {
-            assertFalse(versionValidatorResult.isValid());
-        }
-        assertEquals(expectedVersionValidationCode, versionValidatorResult.getErrorCode());
+    private void validateRecordVersion(final FDBRecordStore store, final Tuple primaryKey, final @Nonnull String expectedVersionValidationCode) {
+        validate(expectedVersionValidationCode, new RecordVersionValidator(store), primaryKey);
     }
 
-    private static void validateRecordValue(final FDBRecordStore store, final Tuple primaryKey, final @Nonnull String expectedValueValidationCode) throws InterruptedException, ExecutionException {
-        RecordValidator valueValidator = new RecordValueValidator(store);
-        final RecordValidationResult valueValidatorResult = valueValidator.validateRecordAsync(primaryKey).get();
+    private void validateRecordValue(final FDBRecordStore store, final Tuple primaryKey, final @Nonnull String expectedValueValidationCode) {
+        validate(expectedValueValidationCode, new RecordValueValidator(store), primaryKey);
+    }
+
+    private void validate(String expectedValueValidationCode, RecordValidator validator, Tuple primaryKey) {
+        RecordValidationResult actualResult = null;
+        try {
+            actualResult = validator.validateRecordAsync(primaryKey).get();
+        } catch (InterruptedException | ExecutionException ex) {
+            throw new RuntimeException("Failed to validate record", ex.getCause());
```

Comment:
Any time you catch an `InterruptedException` you should re-interrupt the thread. 
That being said, if you call `join()` instead of `get()` it doesn't throw `InterruptedException`. Or you could change the method to throw `Exception`, and remove the `catch` here entirely. At least some, if not all of the tests already throw `Exception`.

Reply:
The issue was you cannot throw a checked exception inside a `forEach`. Converting to `join` that only throws unchecked exceptions made this easier.